lib/dao.js

```diff
@@ -2356,6 +2566,184 @@ DataAccessObject.prototype.unsetAttribute = function unsetAttribute(name, nullif
 };
 
 /**
+ * Replace set of attributes.
+ * Performs validation before replacing.
+ *
+ * @trigger `validation`, `save` and `update` hooks
+ * @param {Object} data Data to replace
+ * @param {Object} [options] Options for replace
+ * @param {Function} cb Callback function called with (err, instance)
+ */
+
+DataAccessObject.prototype.replace = function replace(data, options, cb) {
```

Comment:
I'm debating if `replace` should be a prototype method. Maybe a static method such as `replaceById` is more consistent with `updateById` and `destroyById`. If that's agreed, I prefer to use `replace` as an alias to `replaceAll`.